libraries/botframework-connector/src/auth/botFrameworkAuthentication.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Activity, CallerIdConstants } from 'botframework-schema';
+import { AuthenticationConstants } from './authenticationConstants';
+import { ClaimsIdentity } from './claimsIdentity';
+import { ServiceClientCredentials } from '@azure/ms-rest-js';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+import { SkillValidation } from './skillValidation';
+import { assert, Assertion } from 'botbuilder-stdlib';
+
+export interface ProactiveCredentialsResult {
+    credentials: ServiceClientCredentials;
+    scope: string;
+}
+
+export interface AuthenticateRequestResult extends ProactiveCredentialsResult {
+    claimsIdentity: ClaimsIdentity;
+    callerId: string;
+}
+
+export abstract class BotFrameworkAuthentication {
+    static assert: Assertion<BotFrameworkAuthentication> = assert.instanceOf(
+        'BotFrameworkAuthentication',
+        BotFrameworkAuthentication
+    );
+
+    static isType = assert.toTest(BotFrameworkAuthentication.assert);
+
+    /**
+     * Extract app ID from claims identity.
+     *
+     * @param {ClaimsIdentity} claimsIdentity claims identity from which app ID is extracted
+     * @returns {string} the app ID
+     */
+    protected getAppId(claimsIdentity: ClaimsIdentity): string {
```

Comment:
This seemed essentially duplicated from `JwtTokenValidation`, is there any reason not to isolate this logic in one place?